sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/TablesExtensionConfigProvider.cs

```diff
@@ -95,21 +101,64 @@ private static string Resolve(string name, INameResolver nameResolver)
             return nameResolver.ResolveWholeString(name);
         }
 
-        private IAsyncCollector<ITableEntity> BuildFromTableAttribute(TableAttribute attribute)
+        private IAsyncCollector<TableEntity> CreateTableWriter(TableAttribute attribute)
+        {
+            var table = GetTable(attribute);
+
+            return new TableEntityWriter(table, attribute.PartitionKey, attribute.RowKey);
+        }
+
+        private async Task<TableClient> CreateTableClient(TableAttribute attribute, CancellationToken cancellationToken)
         {
             var table = GetTable(attribute);
+            await table.CreateIfNotExistsAsync(cancellationToken).ConfigureAwait(false);
 
-            var writer = new TableEntityWriter<ITableEntity>(table);
-            return writer;
+            return table;
         }
 
-        public ITableEntity JObjectToTableEntityConverterFunc(JObject source, TableAttribute attribute)
+        // Used as an alternative to binding to IQueryable.
+        private async Task<JArray> CreateJArray(TableAttribute attribute, CancellationToken cancellation)
         {
-            if (source == null)
+            var table = GetTable(attribute);
+
+            string filter = attribute.Filter;
+            if (!string.IsNullOrEmpty(attribute.PartitionKey))
+            {
+                var partitionKeyPredicate = TableClient.CreateQueryFilter($"PartitionKey eq {attribute.PartitionKey}");
+                if (!string.IsNullOrEmpty(attribute.Filter))
+                {
+                    filter = $"{partitionKeyPredicate} and {attribute.Filter}";
+                }
+                else
+                {
+                    filter = partitionKeyPredicate;
+                }
+            }
+
+            int? maxPerPage = null;
+            if (attribute.Take > 0)
```

Comment:
I realize this is probably not relevant due to compat, but I think `Take` is misleading here since it does not limit the number of results. So the countRemaining below will limit the number of results incorrectly.

Reply:
>it does not limit the number of results

It does. We use it as the maxPageSizeHint but at the same time do manually counting in 

``` C#
                if (countRemaining == 0)
                {
                    break;
                }
```

Reply:
OK, I see. We effectively bail out of the pageable fetching when we hit the size of the page.